app/models/manageiq/providers/openstack/helper_methods.rb

```diff
@@ -77,11 +72,11 @@ def with_notification(type, options: {})
       success_options.merge!(options)
       error_options.merge!(options)
 
-      # copy subject, initiator and cause from options
+      # extract subject, initiator and cause from options
       named_options_keys = [:subject, :initiator, :cause]
       named_options = {}
       named_options_keys.map do |key|
-        named_options[key] = options.fetch(key, nil)
+        named_options[key] = options.delete(key)
```

Comment:
Wow, we weren't even using it? Crazy.  Nice find.

Reply:
Yeah in fact `options[:subject]` is supposed to be a backup copy of the `subject.name` in the event that the actual subject is deleted.

I'm going to have to check but it looks like this might overwrite the `options[:subject]` object in a before_save hook (though we'd still have an issue with the initiator and cause)

Reply:
Sounds good. I vaguely recall the `subject.name` backup name thing and part of the problem was there were different clients creating these notifications so you'd have to remove passing the subject for each or at least try to normalize them to not use it and only provide a possible identifier for the subject of the notification

Reply:
The subject backup is just taking it from the `subject_type/subject_id` polymorphic assoc so that should work for any/all callers

Reply:
I think this entire thing simplifies to

```diff
-       # copy subject, initiator and cause from options
-       named_options_keys = [:subject, :initiator, :cause]
-       named_options = {}
-       named_options_keys.map do |key|
-         named_options[key] = options.fetch(key, nil)
+       named_options = options.extract!(:subject, :initiator, :cause)
```

Reply:
Okay I checked all of the callers and none of them pass in initiator or cause, only subject and that is changed by the pre-save hook to be a simple name.  I don't think we need a schema migration since there are no notifications which would have serialized AR objects in `options` at least for openstack.